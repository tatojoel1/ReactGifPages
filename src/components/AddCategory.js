import React, { useState } from "react";
import { Form } from "react-bootstrap";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState(""); // ''

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    //console.log("handeSubmit", inputValue);
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <Form.Group controlId="formBasicEmail">
        <p>{inputValue}</p>
        <Form.Label>Type in the box to seek some gifs for you</Form.Label>
        <input
          type="text"
          className="form-control"
          value={inputValue}
          onChange={handleInputChange}
          placeholder="Example...Tacos"
        />
      </Form.Group>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
