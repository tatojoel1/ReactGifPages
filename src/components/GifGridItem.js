import React, { useState, useEffect, useRef } from "react";
import PropTypes from "prop-types";
import { Card, Button, Col, Tooltip, Overlay } from "react-bootstrap";
import { CopyToClipboard } from "react-copy-to-clipboard";

export const GifGridItem = ({ title, url }) => {
  const [copyText, setcopyText] = useState(false);
  const [show, setShow] = useState(false);
  const target = useRef(null);

  useEffect(() => {
    setTimeout(() => {
      setcopyText(false);
      setShow(false);
    }, 2000);
  }, [copyText]);

  return (
    <Col md={4} className="mb-4">
      <Card className="card animate__animated animate__fadeIn">
        <Card.Img variant="top" src={url} alt={title} />
        <Card.Body>
          <Card.Title className="card-title">
            <p>{title}</p>
            {/* {title.length > 0 ? title : "No title found"} */}
          </Card.Title>
        </Card.Body>
        <Card.Text>
          <h4>Link of gif: </h4>
          <a href={url} target="_blank" rel="noopener noreferrer">
            {url}
          </a>
        </Card.Text>
        {/* <Card.Text>{title.length > 0 ? "" : "Sorry, but the author doesn't put it some title in the gif"}</Card.Text> */}
        <Card.Footer>
          <CopyToClipboard text={url} onCopy={() => setcopyText(true)}>
            <Button
              className="copy-button rounded-pill"
              ref={target}
              onClick={() => setShow(!show)}
            >
              Copy
            </Button>
          </CopyToClipboard>
          <Overlay target={target.current} show={show} placement="right">
            {(props) => (
              <Tooltip id="overlay-example" {...props}>
                Copied!
              </Tooltip>
            )}
          </Overlay>
        </Card.Footer>
      </Card>
    </Col>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
