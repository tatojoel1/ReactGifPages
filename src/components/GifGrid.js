import React from "react";
import PropTypes from 'prop-types'
import { CardGroup, Row } from "react-bootstrap";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category);

  return (
    <>
      <CardGroup>
        <h3 className="animate__animated animate__fadeIn"> {category} </h3>

        {loading && <p className="animate__animated animate__flash">Loading</p>}

        <div className="card-grid">
          <Row md={4}>
            {data.map((img) => (
              <GifGridItem key={img.id} {...img} />
            ))}
          </Row>
        </div>
      </CardGroup>
    </>
  );
};

GifGrid.prototypes = {
  category: PropTypes.string.isRequired
}
