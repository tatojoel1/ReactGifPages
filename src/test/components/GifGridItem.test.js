import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

const title = "Un titulo";
const url = "https://localhost/algo.jpg";
const wrapper = shallow(<GifGridItem title={title} url={url} />);

describe("Pruebas en <GifGridItem />", () => {
  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el titulo", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe de tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("CardImg");
    //console.log(img.props());
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe de tener animate__fadeIn", () => {
    const div = wrapper.find("Card");
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
