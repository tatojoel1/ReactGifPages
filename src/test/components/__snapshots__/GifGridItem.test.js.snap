// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<Col
  className="mb-4"
  md={4}
>
  <Card
    body={false}
    className="card animate__animated animate__fadeIn"
  >
    <CardImg
      alt="Un titulo"
      src="https://localhost/algo.jpg"
      variant="top"
    />
    <CardBody>
      <CardTitle
        className="card-title"
      >
        <p>
          Un titulo
        </p>
      </CardTitle>
    </CardBody>
    <CardText>
      <h4>
        Link of gif: 
      </h4>
      <a
        href="https://localhost/algo.jpg"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://localhost/algo.jpg
      </a>
    </CardText>
    <CardFooter>
      <CopyToClipboard
        onCopy={[Function]}
        text="https://localhost/algo.jpg"
      >
        <Button
          active={false}
          className="copy-button rounded-pill"
          disabled={false}
          onClick={[Function]}
          variant="primary"
        >
          Copy
        </Button>
      </CopyToClipboard>
      <Overlay
        placement="right"
        rootClose={false}
        show={false}
        target={null}
        transition={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "appear": false,
              "in": false,
              "mountOnEnter": false,
              "timeout": 300,
              "unmountOnExit": false,
            },
            "displayName": "Fade",
            "render": [Function],
          }
        }
      >
        <Component />
      </Overlay>
    </CardFooter>
  </Card>
</Col>
`;
